meson.build

```diff
@@ -0,0 +1,216 @@
+# NOTE: This build system is maintained by the community.
+project('flatcc',
+  'c', 'cpp',
+  version : '0.6.1',
+  license : 'Apache-2.0',
+  default_options : ['c_std=c11', 'cpp_std=c++11', 'default_library=static'],
```

Comment:
I don't think you need to add `default_library=static` here; the user can configure how they want to do this via a command line option, or Meson will intelligently handle this for you away (i.e. if used as a subproject, Meson will build this as static anyway)

Reply:
Found some time to work on this. So the reason I added `default_library=static` originally is  to make Windows build correctly on the CI. Ultimately without this, Meson's `library()` generates a DLL. Given that flatcc currently does not explicitly export any symbols from this DLL by setting `__declspec(dllexport)` on any symbols, MSVC does not create any import lib `flatcc.lib` causing subsequent links to this library to fail, notably the flatcc cli tool. By forcing the generation of a static library, the link problems are avoided.

I've been trying various ways to get around this including `both_libraries()`, but nothing seems to produce a linkable library on Windows. I think that the best path forward is to leave this as it is `default_library=static` specified in the toplevel Meson script, and if users want to specify `--default-library=shared` on the command-line then they are free to, but with the knowledge that this will cause Windows link failures. This mirrors CMake's `-DBUILD_SHARED_LIBS=on` option, which when activated with flatcc also causes Windows link failures due to missing import libs.

I have a local patch with explicitly adds `__declspec(dllexport)` to flatcc's public API, which seems to resolve the issue for both Meson and CMake, but if such a patch is ever considered for submission to flatcc, I think it would be best to do it separately from this PR. It touches lots of files.

TL;DR: On Windows, neither Meson nor CMake is currently capable of building and linking a shared version of flatcc, so default to static under Meson as is already the case under CMake.

Reply:
Ah OK that makes sense. I ran into a similar issue with nanoarrow last month - the advice I received from upstream was to do something like:

```python
if is_windows
    libtype = 'static_library'
else
    libtype = 'library'
endif

build_target(
    ....,
    target_type: libtype,
)
```

https://github.com/mesonbuild/wrapdb/pull/1536#issuecomment-2136011408

Reply:
Of course could also try to properly export symbols and simulate the issues by setting `-fvisibility=hidden` for gcc, but that would require changes to the flatcc code itself